src/hooks/useSolutionModal/index.tsx

```diff
@@ -0,0 +1,132 @@
+import { Modal } from "@mui/material";
+import { useState } from "react";
+import { PROBLEM_DIFFICULTY_COLOR } from "../../constants/difficulty";
+import { LANGUAGE_COMPILE_MAPPING } from "../../constants/language";
+import CodeBlock from "./CodeBlock";
+import "./styles.css";
+import TabGroup from "./TabGroup";
+
+// hook that returns a function to open the modal, and the modal component
+export default function useSolutionModal() {
+  type LanguageToSolutionMap = Record<string, SolutionAttributes>;
+
+  const [modalOpen, setModalOpen] = useState(false);
+  const [problem, setProblem] = useState<ProblemAttributes | null>(null);
+  const [solutions, setSolutions] = useState<LanguageToSolutionMap>({});
+
+  const solutionLanguages = Object.keys(solutions);
+  const solutionTabs = solutionLanguages.map((language) => {
+    const solutionDetails = solutions[language];
+    const runtime = solutionDetails.runtime_ms;
+    const memory = solutionDetails.memory_usage_mb;
+
+    const hasRuntime = runtime !== null;
+    const hasMemoryUsage = memory !== null;
+
+    return {
+      label: language,
+      content: (
+        <div className="solution-code">
```

Comment:
Why in-line and classes pick one or the other

Reply:
ok